connections.md

```diff
@@ -1,8 +1,252 @@
 ## Connections
 
-- What it means to have a connection.
-- Establishing a connection
-- Ephemeral mode
+A Connection is the practical application of a relationship in DID Communication. Having a connection means that each party in the relationship has a DID for the other parties, and parties can communicate securely using the keys and endpoints within each DID Document.
+
+In order to establish a new connection, we use the OutOfBand Protocol and the DID Exchange Protocol.
+
+
+
+### OutOfBand Protocol
+
+Invitations are usually the first message passed between parties. They are passed out of band, such as in a QR code or a link that can either open in an app or in a browser containing getting started instructions.
+
+The OutOfBand Protocol exists as an independent protocol to allow for the invitation recipient to respond with any of the supported protocols.
+
+
+
+A Public DID functions as a standing invitation. The information within the resolved DID Document serves the same function as the information passed in OutOfBand protocol messages.
+
+TODO: Include details of the protocol.
+
+#### Ephemeral Mode
+
+Some interactions do not need an established connection to complete their purpose. These interactions occur in _ephemeral mode_ and use OutOfBand protocol messages for the bulk of the interaction.
+
+TODO: Include more details about ephemeral mode.
+
+### DID Exchange Protocol
+
+#### 1. Exchange Request
+
+The exchange request message is used to communicate the DID document of the _invitee_ to the _inviter_ using the provisional service information present in the _invitation_ message.
+
+TODO: Clarify that provisioning is only needed if not using an already provisioned DID.
+
+The _invitee_ will provision a new DID according to the DID method spec. For a Peer DID, this involves creating a matching peer DID and key. The newly provisioned DID and DID Doc is presented in the exchange_request message as follows:
+
+TODO: Add clear details about signing the did_doc attachment.
+
+##### Example
+
+```json
+{
+  "@id": "5678876542345",
+  "@type": "https://didcomm.org/didexchange/1.0/request",
+  "~thread": { "pthid": "<id of invitation>" },
+  "label": "Bob",
+  "did": "B.did@B:A",
```

Comment:
I'd prefer to see `did:example:123`, or `did:example:bob` and `example.com` or peer-id, etc...